### DISCUSSION TITLE
(Bi-)weekly bevy dev blog post

### DISCUSSION THREAD

**Discussion by TimJentzsch** [bevyengine/bevy#5725] [Ideas]
# Motivation

Development in Bevy occurs in many areas (e.g. UI, ECS, rendering, etc.) and in many places (PRs, issues, RFCs, different repos) at the same time.
This can make it difficult to get an overview about what is happening, where you could contribute and where help is needed.

# Solution

Add a regularly scheduled dev blog post, that is explicitly targeted at contributors (instead of users of the engine).
This could be hosted on the Bevy website or as a third party tool.

It should be short and just summarize the most "important" developments that happened since the last time-frame, as well as an (automatically generated) list of opened/merged PRs per repository at the end of the post. This makes it easy to digest and relatively easy to write.

I'm not yet sure which time-frame would be most appropriate for this, probably weekly or bi-weekly. Here we should strike a balance between having enough (but not too much) to report in each post and without making them too difficult to maintain. Having regularly scheduled posts like this can be a difficult task in the long term.
Ideally, one contributor has "ownership" of the blog post each time (but it doesn't have to be the same contributor). This person doesn't have to hunt for the needed information in every repository, but should make sure that it is available until submission (for example by asking in the Discord channels if there were any major developments).
Furthermore, this should be an open process. One possibility would be to open a PR at the start of the period with a simple template. Then everyone can comment on the PR to have things added. An alternative would be a separate repository where you can submit PRs with changes.

The content should be focused on engine development, but of course we could also include newly released plugins and games.

I suggest to not advertise this post on platforms such as Reddit, but to keep it internal (e.g. only post the link in Discord) to not "take away hype" from the release blog posts.

# Related Work

Here are some other devlogs that can serve as inspiration:

- [This Week in Rust](https://this-week-in-rust.org/)
- [This Month in Rust GameDev](https://gamedev.rs/news/025/)

**Comment by cart** [comment]
We definitely want this, its just been a matter of investing the time to set up the process and start collecting content.  I was actually planning on picking up this work soonish (after Bevy Jam 2), but if someone else wants to pick it up thats cool too.

Related discussions:
https://github.com/bevyengine/bevy/discussions/4560
https://github.com/bevyengine/bevy-website/issues/179

**Reply by TimJentzsch** [reply]
I would be willing to help with the website implementation and initial collection of content, but I think we first need to answer a couple of open questions:

- How frequently should this be?
- How should it be named? Should the name _not_ include the time frame (This Month In Bevy) to make changes easier in the future?
- Where should it be hosted? Should it be in the "News" tab on the website or in a separate one?
- Who should be the main target? Engine users or developers?
- What kind of content do we want (closely related to the previous question)? Should it focus on engine-internal development or also contain third party tools like plugins and games?
- If it contains third party tools, how do they manually add it or do we do it (covered by the discussion you linked)?

**Reply by cart** [reply]
> How frequently should this be?

I think monthly probably strikes a good balance between frequency and volume of content.

> How should it be named? Should the name not include the time frame (This Month In Bevy) to make changes easier in the future?

I like This Month in Bevy, solely because it follows a pattern that people know and understand in the rust community. We can always rebrand later if needed.

> Where should it be hosted? Should it be in the "News" tab on the website or in a separate one?

At a monthly cadence, in the interest of moving forward quickly I'm cool with including it in the news tab. Any cadence faster than that would probably need its own dedicated section to avoid burying the meaty content.

> What kind of content do we want (closely related to the previous question)? Should it focus on engine-internal development or also contain third party tools like plugins and games?

I think it should include both internal engine development updates and community content like game and plugin updates. (see the template issue I linked above ... lets have this specific conversation in that issue) .

> If it contains third party tools, how do they manually add it or do we do it (covered by the discussion you linked)?

Both: This Month in Bevy curators should populate it with content, which should be supplemented by individual content developers submitting their work. I think a new `#newsletter-submissions` showcase channel has my preference at this point as an easy way for people to submit their content.

Each month we should set up a branch with an empty template that people can submit PRs to. Before releasing each month, curators can create a PR that dumps the `#newsletter-submissions` entries into the template.

**Reply by TimJentzsch** [reply]
My only "concern" with the 1 month cycle is that it might collide away with the 3 month release cycle, take a bit of the hype away from them. But I'm not sure if that would be an actual "issue".

**Reply by Nilirad** [reply]
The 3 month release cycle is not strict, so it doesn't need to be a problem. Also, the newsletter shouldn't be too strict about it.

**Reply by cart** [reply]
> My only "concern" with the 1 month cycle is that it might collide away with the 3 month release cycle, take a bit of the hype away from them. But I'm not sure if that would be an actual "issue".

Any cadence (including numbers of weeks) risks colliding with a release. I think we can time things appropriately when a collision will occur (ex: offset by a week or two).

**Comment by Weibye** [comment]
This is a really good initiative and I something I believe we need. There's so much happening within each section of the engine to the point where it's not really feasible to keep up with all of it unless you are one of few contributors who never seem to sleep. 

I think doing this would allow contributors to get up to speed quicker and reduce a lot uncertainty around "what is happening and where is my help needed?". 

> Who should be the main target? Engine users or developers?

For this reason I believe the primary target-group should be engine development contributors, specifically the "lurkers" who want to contribute but don't do so yet because they feel lost in the vast ocean of "everything happening all at once". 

Secondary target group should be the _active_ contributors where the goal is to make sure the entire team is on the same page.

> How frequently should this be?

Once a month seems fine to start with. Every 2 weeks could make sense once the core team has increased sufficiently in size and / or rate of development increases significantly.

**Comment by Nilirad** [comment]
> The content should be focused on engine development, but of course we could also include newly released plugins and games.

I would prefer, at least in the beginning, to just focus on the official material to keep it simple. We can definitely expand in the future.

**Comment by mockersf** [comment]
Why not do this as part of the existing newsletter "This Month in Rust GameDev"?

The content we want to push (engine update, call to contributions, ecosystem update, ...) would fit in this newsletter, and it would give us a wider reach without having to manage the infrastructure part
